### PR TITLE
CodeCleanup in splashScreen

### PR DESCRIPTION
**Pull request by sidsharma2002**

### CODE REVIEW

File: app/src/main/java/org/greenstand/android/TreeTracker/splash/SplashScreen.kt [hunk 1]
```diff
@@ -80,6 +71,25 @@ fun SplashScreen(
     )
 }
 
+fun isFineAndCoarseLocationPermissionGiven(result: Map<String, Boolean>): Boolean {
```
Comment (Elforama) [inline app/src/main/java/org/greenstand/android/TreeTracker/splash/SplashScreen.kt:74]: Please make these 3 methods private.

File: app/src/main/java/org/greenstand/android/TreeTracker/splash/SplashScreen.kt [hunk 2]
```diff
@@ -36,26 +36,17 @@ fun SplashScreen(
         contract = ActivityResultContracts.RequestMultiplePermissions(),
         onResult = { result ->
             scope.launch {
-                if (result[Manifest.permission.ACCESS_FINE_LOCATION] == true &&
-                    result[Manifest.permission.ACCESS_COARSE_LOCATION] == true
-                ) {
+                if (isFineAndCoarseLocationPermissionGiven(result)) {
```
Comment (Elforama) [inline app/src/main/java/org/greenstand/android/TreeTracker/splash/SplashScreen.kt:39]: I think a less verbose name would do, such as 'isLocationPermissionGranted'

File: app/src/main/java/org/greenstand/android/TreeTracker/splash/SplashScreenViewModel.kt [hunk 3]
```diff
@@ -9,9 +9,9 @@ class SplashScreenViewModel(
     private val users: Users,
 ) : ViewModel() {
 
-    fun migratePreferences() {
+    fun migratePreferencesIfNeeded() {
         preferencesMigrator.migrateIfNeeded()
     }
 
-    suspend fun requiresInitialSetup(): Boolean = users.getPowerUser() == null
+    suspend fun doesRequireInitialSetup(): Boolean = users.getPowerUser() == null
```
Comment (Elforama) [inline app/src/main/java/org/greenstand/android/TreeTracker/splash/SplashScreenViewModel.kt:16]: I'm guessing you updated this to sound more like it returns a boolean, but I think it still sounds like a statement. Can reword this to start with "is"?

**Review by Elforama — CHANGES_REQUESTED** [review]
Thanks for taking the initiative to clean things up! This is looking great, I just added a few comments.

File: app/src/main/java/org/greenstand/android/TreeTracker/splash/SplashScreenViewModel.kt (hunk 3 above)
Comment (sidsharma2002) [reply app/src/main/java/org/greenstand/android/TreeTracker/splash/SplashScreenViewModel.kt:16]: what about 'isInitialSetupRequired' ?

**Review by Elforama — APPROVED** [review]
Thanks for the changes!
